test/picoapp.c

```diff
@@ -450,6 +452,57 @@ int main(int argc, char **argv)
 
         }
         break;
+
+        case '6':
+        {
+            char *nxt, *name = NULL, *area0 = NULL, *area1 = NULL, *dump = NULL;
+            const char pan_addr[] = "2aaa:abcd::0";
+            uint8_t n_id, n_area0, n_area1;
+            struct pico_ip6 pan;
+
+            /* Copy required command line arguments */
+            do {
+                nxt = cpy_arg(&name, optarg);
+                if (!nxt) break;
+                nxt = cpy_arg(&area0, nxt);
+                if (!nxt) break;
+                nxt = cpy_arg(&area1, nxt);
+                if (!nxt) break;
+            } while (0);
```

Comment:
Is this the behaviour of a `goto` statement, hiding under a weird do-while? 
Not sure it wouldn't be more readable by just using plain old goto. "How bad can it be?"

Or does anybody have a better idea?